scipy/fft/_backend.py

```diff
@@ -0,0 +1,141 @@
+import scipy._lib.uarray as ua
+
+from ._basic import (
+    fft, ifft, fft2,ifft2, fftn, ifftn,
+    rfft, irfft, rfft2, irfft2, rfftn, irfftn)
+from ._realtransforms import dct, idct, dst, idst, dctn, idctn, dstn, idstn
+import scipy.fftpack as _fftpack
+from . import _pocketfft
+
+
+class _ScipyBackend:
+    """
+    The default backend for fft calculations
+    """
+    __ua_domain__ = "scipy.fft"
+
+    @staticmethod
+    def __ua_function__(method, args, kwargs):
+        fn = (getattr(_pocketfft, method.__name__, None)
+              or getattr(_fftpack, method.__name__, None))
+
+        if fn is None:
+            return NotImplemented
+        return fn(*args, **kwargs)
+
+
+_named_backends = {
+    'scipy': _ScipyBackend,
+}
+
+def _backend_from_arg(backend):
+    """Maps strings to known backends and validates the backend"""
+
+    if isinstance(backend, str):
+        try:
+            backend = _named_backends[backend]
+        except KeyError:
+            raise ValueError('Unknown backend {}'.format(backend))
+
+    if backend.__ua_domain__ != 'scipy.fft':
+        raise ValueError('Backend does not implement "scipy.fft"')
+
+    return backend
+
+
+def set_global_backend(backend):
+    """Sets the global fft backend
+
+    The global backend has higher priority than registered backends, but lower
+    priority than context-specific backends set with `set_backend`.
+
+    Parameters
+    ----------
+
+    backend: {object, 'scipy'}
+
+        The backend to use.
+        Can either be a ``str`` containing the name of a known backend
+        {'scipy'}, or an object that implements the uarray protocol.
+
+    Raises
+    ------
+
+    ValueError: If the backend does not implement ``scipy.fft``
+
+    Notes
+    -----
+
+    This will overwrite the previously set global backend, which by default is
+    the SciPy implementation.
+
+    """
+
+    backend = _backend_from_arg(backend)
+    ua.set_global_backend(backend)
+
+
+def register_backend(backend):
+    """
+    Register a backend for permanent use.
+
+    Registered backends have the lowest priority and will be tried after the
+    global backend.
+
+    Parameters
+    ----------
+
```

Comment:
There are some extra newlines in the docstrings in this file if you want to get rid of them